Fix test_api_stress.py to accept HTTP 415 (Unsupported Media Type)

## Problem
CI test failing:
```
FAILED backend/tests/test_api_stress.py::test_upload_non_image_file
assert 415 in [400, 422]
```

## Root Cause
- Code quality improvements (PR #34) changed HTTP 400 → 415 for semantic accuracy
- Test file shows it was updated (lines 13-14 have comment about 415)
- BUT line 11 still has old assertion checking for `[400, 422]` instead of just `415`

## Solution
Update line 11 in test_api_stress.py to expect 415 only

## Files to Change
- `backend/tests/test_api_stress.py` line 11

## Acceptance Criteria
- [x] Test expects HTTP 415 (Unsupported Media Type)
- [x] CI passes for test_upload_non_image_file

--- a/backend/tests/test_api_stress.py
+++ b/backend/tests/test_api_stress.py
@@ -17,7 +17,7 @@ def test_upload_non_image_file(client):
     files = {"file": ("test.txt", file_content, "text/plain")}
     
     response = client.post("/api/v1/analyze/image", files=files)
-    
+
     # 415 = Unsupported Media Type (RFC 7231)
     assert response.status_code == 415, (
         f"Expected 415 (Unsupported Media Type) for non-image file, "

